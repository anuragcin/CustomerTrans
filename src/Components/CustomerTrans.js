import React from 'react';
import Table from 'react-bootstrap/Table';
import Badge from 'react-bootstrap/Badge'

const CustomerTrans =(props)=>{
  console.log(props);
  return( <div>
    <Table responsive> 
      <thead>
          <tr>
              <th colSpan="3" width="450px">&nbsp;</th>
              <th>
                <Badge  variant="primary">
                  Total Customer Rewards : {props.totalRewards}
                </Badge>
              </th>
            </tr>
      </thead>
    </Table>
    <Table  responsive striped bordered hover variant="light"> 
      <thead>
            <tr>
              <th>Transaction</th>
              <th>Date</th>
              <th>Amount Spend</th>
              <th>REWARD</th>
              
            </tr>
        </thead>
      {props.data.map((item,i)=>{
          console.log(item)
          
        return (
          <tbody key = {i} id = {i}>
            <tr>
              <td>{item.store}</td>
              <td>{item.date}</td>
              <td>{item.amountSpend}</td>
              <td>{item.points}</td>
            </tr>
          </tbody>
          );
      })}
    </Table>
      <Table > 
      <thead>
          <tr>
              <th colSpan="3" width="450">&nbsp;</th>
              <th>
                <Badge  variant="info">
                  Total Monthly Rewards : {props.totalMonthlyRewards}
                </Badge>
              </th>
            </tr>
      </thead>
      </Table>
    </div>
    
    )

}
  
export default CustomerTrans;