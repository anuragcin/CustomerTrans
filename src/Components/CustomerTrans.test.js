import React from 'react';
import {shallow} from 'enzyme';
import CustomerTrans from './CustomerTrans';


it('expect to render CustomerTrans Component',()=>{     
      const mockData = [
        {
            amountSpend: 120,
            date: "12/1/2019",
            firstName: "John",
            lastName: "Doe",
            month: 11,
            points: 90,
            store: "Walmart",
        }
      ];
      const mockTotalMonthlyRewards=90;
      const mockTotalRewards=220;

    expect(shallow(
    <CustomerTrans 
        data={mockData}
        totalMonthlyRewards = {mockTotalMonthlyRewards}
        totalRewards = {mockTotalRewards}
    />)).toMatchSnapshot();
})