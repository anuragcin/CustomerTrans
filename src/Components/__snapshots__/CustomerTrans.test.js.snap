// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CustomerTrans Component 1`] = `ShallowWrapper {}`;
